doc/tutorials/computing-product-analytics.md

```diff
@@ -0,0 +1,258 @@
+| 2      | 4 |3|5|
+| 3      | 5 |1|3|
+| 3      | 5 |2|4|
+| 3      | 5 |3|5|
+
++++
+
+Then, with **WHERE c1.month = c2.month - 1**, we filter out the total number of users for subsequential months. The expected output is: 
+| c1.month | c1.total_users| c2.month| c2.total_users|
+|--------|------------|--------|------------|
+| 1      | 3|2      | 4 |
+| 2      | 4 |3|5|
+
+As shown above, we calculate the final growth rate using c1.total_users and c2.total_users.
+
++++
+
+### Retention
+
++++
+
+Retention is a bit more complex. Every company might have a different definition of user retention. Here, we defiend retention as the percentage of users who still use the app one month after their first login.
+
++++
+
+We first retrieve two tables. The first one is the number of first-login users in each month. We will create a temporary table called first_time_users to hold the results.
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%%sql
+DROP TABLE IF EXISTS first_time_users;
+CREATE TABLE first_time_users (
+  month INT,
+  first_time_users INT
+);
+
+INSERT INTO first_time_users (month, first_time_users)
+SELECT MONTH(date) AS month, COUNT(DISTINCT u.user_id) AS first_time_users
+FROM user_activity u
+INNER JOIN (
+  SELECT user_id, MIN(date) AS first_login
+  FROM user_activity
+  GROUP BY user_id
+) t ON u.user_id = t.user_id AND u.date = t.first_login
+GROUP BY MONTH(date);
+
+SELECT * FROM first_time_users
+```
+
+This means that in January, 3 users start to use the app. Similarly, 2 start using in Febuary and 2 start using in March.
+
++++
+
+Then we check the retention users (number of users who still use the app after one month of first-login) in each month.
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%%sql
+DROP TABLE IF EXISTS retention_users;
+CREATE TABLE retention_users (
+  month INT,
+  retention_users INT
+);
+
+INSERT INTO retention_users (month, retention_users)
+SELECT MONTH(first_login) AS month, COUNT(DISTINCT u. user_id) AS retention_users
+FROM user_activity u
+INNER JOIN (
+SELECT user_id, MIN(date) AS first_login
+FROM user_activity
+GROUP BY user_id) t 
+ON u.user_id = t.user_id
+WHERE MONTH(date) = MONTH(first_login) +1
+GROUP BY MONTH(first_login);
+SELECT * FROM retention_users
+```
+
+Here, the code that `WHERE MONTH(date) = MONTH(first_login) + 1` make sure that we only take users who still used the app at least one month behind into account. 
+Only 2 users out of 3 continue to use the app after a month.
+
++++
+
+Then, we combine these two table and make our calculation
+
+We join the new_users and retention users based on the month we are looking at. Then, we make calculation for retention_rate.
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%%sql
+SELECT f.month, first_time_users, IFNULL(retention_users, 0) AS retention_users, ROUND(retention_users * 1.0 / first_time_users, 4)*100 AS retention_rate
+FROM first_time_users f 
+FULL OUTER JOIN retention_users r
+ON f.month = r.month
```

Comment:
Suggestion:

Add a summary of what was learned in this tutorial, how JupySQL was used to achieve the results and how it helped you streamline the analysis. Point to next steps if applicable.